api/v1/mdb/mongodb_validation.go

```diff
@@ -137,6 +138,52 @@ func oidcAuthValidators(db DbCommonSpec) []func(DbCommonSpec) v1.ValidationResul
 	return validators
 }
 
+// oidcProviderConfigUniqueIssuerURIValidation is based on the documentation here:
+// https://www.mongodb.com/docs/manual/reference/parameters/#oidcidentityproviders-fields
+func oidcProviderConfigUniqueIssuerURIValidation(configs []OIDCProviderConfig) func(DbCommonSpec) v1.ValidationResult {
+	return func(d DbCommonSpec) v1.ValidationResult {
+		if len(configs) == 0 {
+			return v1.ValidationSuccess()
+		}
+
+		// Check if version supports duplicate issuers (7.0, 7.3, or 8.0+)
+		versionParts := strings.Split(strings.TrimSuffix(d.Version, "-ent"), ".")
+		supportsMultipleIssuers := false
+		if len(versionParts) >= 2 {
+			major := versionParts[0]
+			minor := versionParts[1]
+			if major == "8" || (major == "7" && (minor == "0" || minor == "3")) {
+				supportsMultipleIssuers = true
+			}
+		}
+
+		if supportsMultipleIssuers {
+			// Track issuer+audience combinations
+			issuerAudienceCombos := make(map[string]string)
+			for _, config := range configs {
+				comboKey := config.IssuerURI + ":" + config.Audience
+				if previousConfig, exists := issuerAudienceCombos[comboKey]; exists {
+					return v1.ValidationWarning("OIDC provider configs %q and %q have duplicate IssuerURI and Audience combination",
```

Comment:
nit: we will only get warning or error for single issue. We should combine all errors or warnings together if there are more then one. This is hard to do with current validation mechanism unfortunately. Just wanted to highlight this problem.